lib/block-supports/background.php

```diff
@@ -30,6 +30,45 @@ function gutenberg_register_background_support( $block_type ) {
 	}
 }
 
+/**
+ * Given a theme.json or block background styles, returns the background styles for a block.
+ *
+ * @param  array  $background_styles Background style properties.
+ * @return array                     Style engine array of CSS string and style declarations.
+ */
+function gutenberg_get_background_support_styles( $background_styles = array() ) {
+	$background_image_source = $background_styles['backgroundImage']['source'] ?? null;
+
+	/*
+	 * If the background image is a string, it's a URL.
+	 * Assume "file" source.
+	 */
+	if ( is_string( $background_styles['backgroundImage'] ) ) {
+		$url                                  = esc_url( $background_styles['backgroundImage'] );
+		$background_image_source              = 'file';
+		$background_styles['backgroundImage'] = array(
+			'url' => $url,
+		);
+	}
+
+	if ( ! empty( $background_styles['backgroundImage']['url'] ) ) {
```

Comment:
Wondering if this is the right place for value coercion? Is it beneficial to provide defaults for incoming theme.json backgroundImage properties?

Or should we let theme.json values stand alone?

Reply:
Just a drive-by comment: it looks like the right place to me, since it's fairly close to where the styles are being output, and it's nice for everything to be in the one place.

> Is it beneficial to provide defaults for incoming theme.json backgroundImage properties?

I was imagining that the defaults would be applied the same for `theme.json` properties as for an individual block post content — that way folks don't need to know to set the size to `cover`, it can just work that way consistently out of the box. Also, it means once we add the UI controls, they can work the same in global styles and in the post editor (from memory there'll be a bit of refactoring required of the JS so that the components can be shared across both areas). It's not a strongly held opinion, though!

Reply:
Thanks for the sanity check @andrewserong 

Agree that consistency between block supports and global styles would be a positive thing! 👍🏻